ext/math.go

```diff
@@ -86,28 +87,311 @@ import (
 //	math.least('string') // parse error
 //	math.least(a, b)     // check-time error if a or b is non-numeric
 //	math.least(dyn('string')) // runtime error
+//
+// # Math.BitOr
+//
+// Introduced at version: 1
+//
+// Var-args macro which accepts a variable quantity of [1, 32) integers and
+// performs a bitwise-OR operation on all of them. The macro is expanded into
+// a call to the internal-only overload of math.@bitOr(number),
+// math.@bitOr(number, number), or math.@bitOr(list)
+//
+// The numbers must be signed or unsigned integers, and for list based
+// arguments the numbers must all be of the same type. An emtpy list provided
+// as input will be a runtime error.
+//
+//	math.bitOr(<int>, <int>) -> <int>
+//	math.bitOr(<uint>, <uint>) -> <uint>
+//
+// Examples:
+//
+//	math.bitOr(1u, 2u)    // returns 3u
+//	math.bitOr(-2, -4)    // returns -2
+//
+// # Math.BitAnd
+//
+// Introduced at version: 1
+//
+// Performs a bitwise-AND operation over two int or uint values.
+//
+//	math.bitAnd(<int>, <int>) -> <int>
+//	math.bitAnd(<uint>, <uint>) -> <uint>
+//
+// Examples:
+//
+//	math.bitAnd(3u, 2u)   // return 2u
+//	math.bitAnd(3, 5)     // returns 3
+//	math.bitAnd(-3, -5)   // returns -7
+//
+// # Math.BitXor
+//
+// Introduced at version: 1
+//
+//	math.bitXor(<int>, <int>) -> <int>
+//	math.bitXor(<uint>, <uint>) -> <uint>
+//
+// Performs a bitwise-XOR  operation over two int or uint values.
+//
+// Examples:
+//
+//	math.bitXor(3u, 5u) // returns 6u
+//	math.bitXor(1, 3)   // returns 2
+//
+// # Math.BitNot
+//
+// Introduced at version: 1
+//
+// Function which accepts a single int or uint and performs a bitwise-NOT
+// ones-complement of the given binary value.
+//
+//	math.bitNot(<int>) -> <int>
+//	math.bitNot(<uint>) -> <uint>
+//
+// Examples
+//
+//	math.bitNot(1)  // returns -1
+//	math.bitNot(-1) // return 0
+//	math.bitNot(0u) // returns 18446744073709551615u
+//
+// # Math.BitShiftLeft
+//
+// Introduced at version: 1
+//
+// Perform a left shift of bits on the first parameter, by the amount of bits
+// specified in the second parameter. The first parameter is either a uint or
+// an int. The second parameter must be an int.
+//
+// When the second parameter is 64 or greater, 0 will be always be returned
+// since the number of bits shifted is greater than or equal to the total bit
+// length of the number being shifted.
+//
+//	math.bitShiftLeft(<int>, <int>) -> <int>
```

Comment:
Should we just define the second parameter to be uint? In most languages, negative shift is either undefined or an error.

Reply:
Within ZetaSQL, the shift argument is an int. I kept it for parity there.

Reply:
We should document this behavior, this will have to be special cased in both Java/C++ to produce an error

Reply:
Oh, I see you mean documenting the error case for a negative shift?

Reply:
Yep, negative shifts isn't clearly established across most languages (It either results in an error, always returns 0, shift direction is flipped, with ambiguity introduced between logical/arithmetic shifts etc.).

Reply:
Done. I'll add conformance tests as well.